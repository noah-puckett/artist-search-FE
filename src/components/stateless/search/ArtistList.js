import React from 'react';
import PropTypes from 'prop-types';
import Artist from '../search/Artist';
import { Link } from 'react-router-dom';

function ArtistList({ artists }) {
    const artistsArray = artists.map(artist => {
        return (
            <li style={{ listStyleType: 'none' }} key={artist.id}>
                <Link to={`/${artist.name}/${artist.id}/albums`}>
                    <Artist name={artist.name} id={artist.id} />
                </Link>
            </li>
        );
    });
    return artistsArray;
}

ArtistList.propTypes = {
    artists: PropTypes.array.isRequired,
};

export default ArtistList;
