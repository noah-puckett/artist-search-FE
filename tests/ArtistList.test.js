import React from 'react';
import { shallow } from 'enzyme';
import ArtistList from '../src/components/stateless/search/ArtistList';

describe('ArtistList component', () => {
    it('renders ArtistList', () => {
        const wrapper = shallow(<ArtistList artists={[]} />);
        expect(wrapper).toMatchSnapshot();
    });
});
