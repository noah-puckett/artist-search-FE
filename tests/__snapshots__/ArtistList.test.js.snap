// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistList component renders ArtistList 1`] = `null`;
